x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx

```diff
@@ -687,6 +712,26 @@ export class VectorLayer extends AbstractLayer implements IVectorLayer {
     }
   }
 
+  async _syncIsEditable({ syncContext }: { syncContext: DataRequestContext }) {
```

Comment:
Thanks for moving this into a data request. This is much better. One more thing, can you also check that isEditable is only loaded when the map is not in readOnly mode? That way, this request will not occur when map is embedded in dashboard and canvas where editing is not possible.

You will need to add reduxStore.ui.isReadOnly to MapFilters